File: 2020/Day2Test.cs
using System;
using NUnit.Framework;

namespace AoC.Tests
{
    [TestFixture]
    class Day2Test
    {
        public static readonly string[] INPUT = { 
            "1-3 a: abcde",
            "1-3 b: cdefg",
            "2-9 c: ccccccccc"
        };
 
        private IPuzzle puzzle;

        public Day2Test()
        {
            this.puzzle = new Day2();
        }

        [TestCase]
        public void TestPart1()
        {   
            Assert.AreEqual("2", this.puzzle.Part1(INPUT));
        }
        
        [TestCase]
        public void TestPart2()
        {   
            Assert.AreEqual("1", this.puzzle.Part2(INPUT));
        }
    }
}

File: aoc2025/cmd/day2/day2_test.go
package main

import (
	"testing"
)

var input = `11-22,95-115,998-1012,1188511880-1188511890,222220-222224,1698522-1698528,446443-446449,38593856-38593862,565653-565659,824824821-824824827,2121212118-2121212124`

func TestPart1(t *testing.T) {
	actual := Part1(input)
	expected := "1227775554"
	if actual != expected {
		t.Errorf(`Part1(input) = %q, want %q`, actual, expected)
	}
}
func TestPart2(t *testing.T) {
	actual := Part2(input)
	expected := "4174379265"
	if actual != expected {
		t.Errorf(`Part2(input) = %q, want %q`, actual, expected)
	}
}
